Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/jxbrowserutils/BrowserUtils.java

```diff
@@ -0,0 +1,273 @@
+            in.close();
+
+            // read md5 file to string
+            String md5FileContent = FileUtils.readFileToString(md5File, "UTF-8");
+            return Objects.equal(md5FileContent, md5GeneratedByJar);
+        } finally {
+            if (in != null) {
+                in.close();
+            }
+            if (scanner != null) {
+                scanner.close();
+            }
+        }
+    }
+
+    public static void downloadFromAzure(String fileName) throws Exception {
+        // connect azure
+        CloudStorageAccount account = CloudStorageAccount.parse(storageConnectionString);
+        CloudBlobClient serviceClient = account.createCloudBlobClient();
+        // get container
+        CloudBlobContainer container = serviceClient.getContainerReference("libcontainer1");
+        // download the file
+        CloudBlockBlob blob = container.getBlockBlobReference(fileName);
+        File downloadingFile = new File(fileName);
+        blob.downloadToFile(downloadingFile.getAbsolutePath());
+        logger.info("download " + fileName + " complete.");
+    }
+
+    public static void downloadAndCheckJarHelper(ListeningExecutorService threadPool, String downloadFilePrefix, int checkMd5RetryTimes, int downloadRetryTimes) throws Exception {
+        if (downloadFilePrefix == null) {
+            throw new MyException("file name cannot be null.");
+        }
+        logger.info("download file prefix: " + downloadFilePrefix);
+        String jarName = downloadFilePrefix + ".jar";
+        String md5Name = downloadFilePrefix + ".md5";
+
+        ListenableFuture<Boolean> future = threadPool.submit(() -> {
+            downloadFromAzure(md5Name);
+            downloadFromAzure(jarName);
+            return true;
+        });
+
+        Futures.addCallback(future, new FutureCallback<Boolean>() {
+            @Override
+            public void onSuccess(Boolean aBoolean) {
+                try {
+                    logger.info("future: " + jarName + " onSuccess.");
+                    if (!md5Check(jarName, md5Name)) {
+                        logger.info("MD5 check for " + jarName + " failed.");
+                        FileUtils.deleteQuietly(new File(jarName));
+                        FileUtils.deleteQuietly(new File(md5Name));
+                        // retry downloading and checking md5.
+                        if (checkMd5RetryTimes < MAX_CHECK_FAILURE_RETRY_TIMES) {
+                            downloadAndCheckJarHelper(threadPool, downloadFilePrefix, checkMd5RetryTimes + 1, downloadRetryTimes);
+                        } else {
+                            throw new MyException("Retry " + MAX_CHECK_FAILURE_RETRY_TIMES + " times, still failed in checking MD5 of " + jarName);
+                        }
+                    } else {
+                        logger.info(jarName + " pass MD5 checking.");
+                    }
+                } catch (Exception e) {
+                    e.printStackTrace();
```

Comment:
>e.printStackTrace(); [](start = 20, length = 20)

Log this exception and show them on IDE? Or maybe we can provide some useful information for customers.